yt_dlp/extractor/cnn.py

```diff
@@ -1,146 +1,225 @@
+            'info_dict': {
+                'id': 'med04507d8ca3da827001f63d22af321ec29c7d97b',
+                'ext': 'mp4',
+                'title': '\'Wise\': Buttigieg on Harris\' handling of interview question about gender transition surgery',
+                'display_id': 'pete-buttigieg-harris-fox-newssrc-digvid',
+                'description': 'md5:602a8a7e853ed5e574acd3159428c98e',
+                'thumbnail': 'https://media.cnn.com/api/v1/images/stellar/prod/buttigieg-20241017040412074.jpg?c=original',
+                'duration': 145.0,
+                'timestamp': 1729137765.846,
+                'upload_date': '20241017',
+                'modified_timestamp': 1729138184.891,
+                'modified_date': '20241017',
+            },
+            'params': {'format': 'direct'},
+        }],
+    }]
 
     def _real_extract(self, url):
-        sub_domain, path, page_title = self._match_valid_url(url).groups()
-        if sub_domain not in ('money', 'edition'):
-            sub_domain = 'edition'
-        config = self._CONFIG[sub_domain]
-        return self._extract_cvp_info(
-            config['data_src'] % path, page_title, {
-                'default': {
-                    'media_src': config['media_src'],
-                },
-                'f4m': {
-                    'host': 'cnn-vh.akamaihd.net',
-                },
-            })
+        display_id = self._match_valid_url(url).group('display_id')
+        webpage = self._download_webpage(url, display_id)
+        app_id = traverse_obj(
+            self._search_json(r'window\.env\s*=', webpage, 'window env', display_id, default={}),
+            ('TOP_AUTH_SERVICE_APP_ID', {str}))
 
+        entries = []
+        for player_data in traverse_obj(webpage, (
+                {find_elements(tag='div', attr='data-component-name', value='video-player', html=True)},
+                ..., {extract_attributes}, all, lambda _, v: v['data-media-id'] and v['data-video-resource-parent-uri'])):
+            media_id = player_data['data-media-id']
+            parent_uri = player_data['data-video-resource-parent-uri']
+            formats, subtitles = [], {}
 
-class CNNBlogsIE(InfoExtractor):
-    _VALID_URL = r'https?://[^\.]+\.blogs\.cnn\.com/.+'
-    _TEST = {
-        'url': 'http://reliablesources.blogs.cnn.com/2014/02/09/criminalizing-journalism/',
-        'md5': '3e56f97b0b6ffb4b79f4ea0749551084',
-        'info_dict': {
-            'id': 'bestoftv/2014/02/09/criminalizing-journalism.cnn',
-            'ext': 'mp4',
-            'title': 'Criminalizing journalism?',
-            'description': 'Glenn Greenwald responds to comments made this week on Capitol Hill that journalists could be criminal accessories.',
-            'upload_date': '20140209',
-        },
-        'expected_warnings': ['Failed to download m3u8 information'],
-        'add_ie': ['CNN'],
-    }
+            video_data = self._download_json(
```

Comment:
then we'd have to indent all of this, which is making me dislike my own idea
```suggestion
            if parent_uri:
                video_data = self._download_json(
```